IANotificationsTestProject/In App Notifications/NoticeController.swift

```diff
@@ -0,0 +1,65 @@
+import UIKit
+
+class NoticeController {
+    private var notices: [Notice] = []
+    private let noticePresenter = NoticePresenter()
+
+    private var activeViewIsBeingTouched: Bool = false
+
+    func present(_ notice: Notice) {
+        if noticePresenter.isPresenting {
+            notices.append(notice)
+        } else {
+            let noticeView = makeNoticeView(from: notice)
+
+            noticePresenter.presentNoticeView(noticeView) {
+                self.dismiss(noticeView)
+            }
+        }
+    }
+
+    func dismiss(_ noticeView: NoticeView) {
+        guard !activeViewIsBeingTouched else {
+            DispatchQueue.main.asyncAfter(deadline: .now() + Times.tapDelay) {
+                self.dismiss(noticeView)
+            }
+            return
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + Times.tapDelay) {
+            self.noticePresenter.dismissNotification {
+                if !self.notices.isEmpty {
+                    self.present(self.notices.removeFirst())
+                }
+            }
+        }
```

Comment:
There's a bit of code duplication between L23 and L28

Reply:
good catch, this was resolved by changing how the taps are responded to